pallets/pools/src/lib.rs

```diff
@@ -545,21 +570,93 @@ pub mod pallet {
+				EpochExecution::<T>::try_get(pool_id).is_err(),
+				Error::<T>::InSubmissionPeriod
 			);
 
-			Pool::<T>::try_mutate(pool_id, |pool| -> DispatchResult {
-				let pool = pool.as_mut().ok_or(Error::<T>::NoSuchPool)?;
+			if let Change::NewValue(min_epoch_time) = changes.min_epoch_time {
+				ensure!(
+					min_epoch_time >= T::MinEpochTimeLowerBound::get()
+						&& min_epoch_time <= T::MinEpochTimeUpperBound::get(),
+					Error::<T>::PoolParameterBoundViolated
+				);
+			}
+
+			if let Change::NewValue(max_nav_age) = changes.max_nav_age {
+				ensure!(
+					max_nav_age <= T::MaxNAVAgeUpperBound::get(),
+					Error::<T>::PoolParameterBoundViolated
+				);
+			}
+
+			let pool = Pool::<T>::try_get(pool_id).map_err(|_| Error::<T>::NoSuchPool)?;
+
+			if let Change::NewValue(tranches) = &changes.tranches {
+				Self::is_valid_tranche_change(Some(&pool.tranches), &tranches)?;
+			}
+
+			if T::MinUpdateDelay::get() == 0
+				&& T::RequireRedeemFulfillmentsBeforeUpdates::get() == false
+			{
+				Self::do_update_pool(&pool_id, &changes)
+			} else {
+				// If an update was already stored, this will override it
+				ScheduledUpdate::<T>::insert(
+					pool_id,
+					ScheduledUpdateDetails {
+						changes: changes.clone(),
+						scheduled_time: Self::now().saturating_add(T::MinUpdateDelay::get()),
+					},
+				);
 
-				pool.parameters.min_epoch_time = min_epoch_time;
-				pool.parameters.challenge_time = challenge_time;
-				pool.parameters.max_nav_age = max_nav_age;
-				Self::deposit_event(Event::Updated(pool_id));
 				Ok(())
-			})
+			}
+		}
+
+		/// Executed a scheduled update to the pool.
+		///
+		/// This checks if the scheduled time is in the past
+		/// and, if required, if there are no outstanding
+		/// redeem orders. If both apply, then the scheduled
+		/// changes are applied.
+		#[pallet::weight(200_000_000)]
+		pub fn execute_scheduled_update(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			let update = ScheduledUpdate::<T>::try_get(pool_id)
+				.map_err(|_| Error::<T>::NoScheduledUpdate)?;
+
+			ensure!(
+				Self::now() >= update.scheduled_time,
+				Error::<T>::ScheduledTimeHasNotPassed
+			);
+
+			if T::RequireRedeemFulfillmentsBeforeUpdates::get() == true {
+				let pool = Pool::<T>::try_get(pool_id).map_err(|_| Error::<T>::NoSuchPool)?;
+
+				// The epoch in which the redemptions were fulfilled,
+				// should have closed after the scheduled time already,
+				// to ensure that investors had the `MinUpdateDelay`
+				// to submit their redemption orders.
+				ensure!(
+					Self::now() >= pool.epoch.last_closed,
+					Error::<T>::ScheduledTimeHasNotPassed
+				);
+
+				ensure!(
+					pool.tranches.acc_outstanding_redemptions()?.is_zero(),
+					Error::<T>::RedemptionsHaveNotBeenFulfilled
+				);
+
+				Self::do_update_pool(&pool_id, &update.changes)?;
+			} else {
+				Self::do_update_pool(&pool_id, &update.changes)?;
```

Comment:
What about getting the `PoolDetails` beforehand and erroring out if the pool does not exists and pass it then down the `do_update_pool` logic?

Reply:
This would also mean moving `try_mutate` to the places where this is called, which I think is adding rather than reducing code.